doc/platform/common_config.md

```diff
@@ -0,0 +1,107 @@
+
+The design change for syncd_init_common.sh is targeted on the config_syncd_bcm() which is only for BRCM switch chip. In the config_syncd_bcm() function, it will load the common silicon config.bcm file by looking for a *.bcm file in the  /usr/share/sonic/device/x86_64-broadcom_common/broadcom-sonic-{$chip_id} folder where the $chip_id is the upper 3 nibbles of the switching silicon's device ID as listed in SDK/include/soc/devids.h. Then, merge common  broadcom-sonic-{$chip_id}.config.bcm  file with the existing platform specific config.bcm file in which the duplicate configuration entries in the platform specific file will override entries in the common broadcom-sonic-{$chip_id}.config.bcm file. 
+
+Since the platform specific config.bcm is read only in docker, the design copies the platform specific config.bcm and sai.profile to /tmp for handling common config merge process. The /tmp/sai.profile will be modified to point to the merged config.bcm under/tmp directory. The following switch initialization will reference to the new merged config.bcm pointed by updated sai.profile from the /tmp directory.
+
+
+## 4 Serviceability and DEBUG
+### 4.1 Syslogs
+During system booting syncd initialization, the Per-switching silicon common config design will log the common config merged information to the syslogs for identifying which common config be merged to the ODM config.bcm for syncd initialization.
+
+### Examples:
+```
+Nov  3 09:19:25.373964 2020 sonic INFO syncd#supervisord: syncd Merging 
+/usr/share/sonic/hwsku/td3-ix8a-bwde-48x25G+8x100G.config.bcm with
+/usr/share/sonic/device/x86_64-broadcom_common/x86_64-broadcom_b77
+/broadcom-sonic-td3.config.bcm, merge files stored in 
+/tmp/td3-ix8a-bwde-48x25G+8x100G.config.bcm
+```
+### 4.2 Debug logs
+The finial config.bcm and sai.profile will copy to the shared folder /var/run/sswsyncd/ in the syncd docker for 'show tech' debugging purpose. Developer can check the final config.bcm file for debugging from the show tech dump /sai/xxx.config.bcm.
+
+### Examples:
+- Issue 'show tech'
+- Check sonic_dump_sonic_20201103_090412/sai/td3-ix8-48x25G+8x100G.config.bcm  
+```
+parity_correction=1  
+flowtracker_enable=2  
+flowtracker_max_flows=48000  
+flowtracker_drop_monitor_enable=1  
+flowtracker_export_interval_usecs=1000000  
+flowtracker_max_export_pkt_length=9000  
+flowtracker_fsp_reinject_max_length=128  
+num_queues_pci=46  
+num_queues_uc0=1  
+num_queues_uc1=1  
+sai_eapp_config_file=/etc/broadcom/eapps_cfg.json  
+```
+
+## 5 Scalability
+NA
+## 6 Unit Test
+- Check the system status to make sure syncd initial success
+- Check the syslog to make sure the common config merged to
+   config.bcm successfully
+- Check the final merged config.bcm dump from show tech dump 
+
```

Comment:
Do you have plans to extract those "common" soc properties from each of the platform specific config.bcm file into the common config.bcm file?

Reply:
No, we don't have the plans to merge anything common from platform specific config.bcm to common config.bcm.
It will break our rule. In the design, when we find the same common config property existing on both platform specific and common specific. We will keep the original config in platform specific without merging it.